### PR TITLE
Add signature and publickey in EncryptionKey

### LINKED ISSUE #2108: [Talos] There is information to be added when providing the encryption key
Votera app receives a VoterCard from Talos when it log-in.
Votera app receives an encryption key from Talos before voting.

Votera app encrypts VoterCard and stores it in the app for ease of use.
When a user operates multiple nodes and receives an encryption key from Talos, it is not possible to determine which node the encryption key belongs to.

In addition, I think a safer voting system can be provided if a method is provided to verify whether the data received via QR code is accurately delivered.

### PR DESCRIPTION
**Pull request by TrustHenry**
Votera app encrypts VoterCard and stores it in the app for ease of use.
When a user operates multiple nodes and receives an encryption key from Talos, it is not possible to determine which node the encryption key belongs to.

In addition, I think a safer voting system can be provided if a method is provided to verify whether the data received via QR code is accurately delivered.

Related to #2108

### CODE REVIEW

File: source/agora/node/admin/AdminInterface.d [hunk 1]
```diff
@@ -317,6 +319,52 @@ public struct EncryptionKey
 
     // Encryption key
     public Hash value;
+
+    // Validator that this encryption key
```
Comment (linked0) [inline source/agora/node/admin/AdminInterface.d]:
```suggestion
    // Validator that owns this encryption key
```

**Review by linked0 — APPROVED** [review]
LGTM, one nit.

File: source/agora/node/admin/AdminInterface.d (hunk 1 above)
Comment (TrustHenry) [reply source/agora/node/admin/AdminInterface.d]: Thanks, I fixed it now.

**Comment by TrustHenry** [comment]
This PR can merge before PR #2113

**Review by kchulj — APPROVED** [review]
LGTM

File: source/agora/node/admin/AdminInterface.d [hunk 2]
```diff
@@ -193,8 +193,10 @@ public class AdminInterface : NodeControlAPI
         EncryptionKey encryptionKey = EncryptionKey(
             app,
             Height(height),
-            key
+            key,
+            this.key_pair.address,
         );
+        encryptionKey.signature = this.key_pair.secret.sign(hashFull(encryptionKey)[]);
```
Comment (Geod24) [inline source/agora/node/admin/AdminInterface.d]:
```suggestion
        encryptionKey.signature = this.key_pair.secret.sign(encryptionKey);
```

File: source/agora/node/admin/AdminInterface.d [hunk 3]
```diff
@@ -193,8 +193,10 @@ public class AdminInterface : NodeControlAPI
         EncryptionKey encryptionKey = EncryptionKey(
             app,
             Height(height),
-            key
+            key,
+            this.key_pair.address,
```
Comment (Geod24) [inline source/agora/node/admin/AdminInterface.d:197]: Note that you are still hashing `Hash.init` (see L188)

File: source/agora/node/admin/AdminInterface.d [hunk 4]
```diff
@@ -317,6 +319,52 @@ public struct EncryptionKey
...
+        Params:
+            dg = hashing function
+
+    ***************************************************************************/
+
+    public void computeHash (scope HashDg dg) const nothrow @safe @nogc
+    {
+        hashPart(this.app, dg);
+        hashPart(this.height.value, dg);
+        hashPart(this.value[], dg);
```
Comment (Geod24) [inline source/agora/node/admin/AdminInterface.d]:
```suggestion
        hashPart(this.value, dg);
```

**Review by Geod24 — CHANGES_REQUESTED** [review]

File: source/agora/node/admin/AdminInterface.d (hunk 3 above)
Comment (TrustHenry) [reply source/agora/node/admin/AdminInterface.d:197]: https://github.com/bosagora/agora/pull/2113
It is being applied by other PR.
If it is merged, I will rebase the PR.

File: source/agora/node/admin/AdminInterface.d (hunk 2 above)
Comment (TrustHenry) [reply source/agora/node/admin/AdminInterface.d]: Why? Don't you sign the hash of the encryption key?
The BOA-SDK we provide provides a hash function, but no serialization is implemented.
@Geod24 Could you tell me why?

Comment (Geod24) [reply source/agora/node/admin/AdminInterface.d]: We perform the hash in the library itself: https://github.com/bosagora/crypto/blob/ba1eb6b34e8cb680177ea07b7a151d85d5222291/source/agora/crypto/Schnorr.d#L218-L242
But it is hashed with other components.

Comment (TrustHenry) [reply source/agora/node/admin/AdminInterface.d]: Thanks, I updated.

File: source/agora/node/admin/AdminInterface.d (hunk 4 above)
Comment (TrustHenry) [reply source/agora/node/admin/AdminInterface.d]: I updated.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**